### PR TITLE
python310Packages.unrpa: specify passthru.optional-dependencies

### LINKED ISSUE #218108: Build failure: python38Packages.meson
### Steps To Reproduce

1. build `python38Packages.meson`

### Build log

[python38Packages.meson.log](https://github.com/NixOS/nixpkgs/files/10829336/python38Packages.meson.log)

```
[1/2] Compiling C object ext/tachyon.so.p/tachyon_module.c.o
FAILED: ext/tachyon.so.p/tachyon_module.c.o 
gcc -Iext/tachyon.so.p -Iext '-I../test cases/python3/2 extmodule/ext' -I/nix/store/jb43gwp472xqwggcwjdvnc4lj26f8x6a-python3-3.8.16/include/python3.8 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O3 -fPIC -MD -MQ ext/tachyon.so.p/tachyon_module.c.o -MF ext/tachyon.so.p/tachyon_module.c.o.d -o ext/tachyon.so.p/tachyon_module.c.o -c '../test cases/python3/2 extmodule/ext/tachyon_module.c'
In file included from ../test cases/python3/2 extmodule/ext/tachyon_module.c:19:
/nix/store/jb43gwp472xqwggcwjdvnc4lj26f8x6a-python3-3.8.16/include/python3.8/Python.h:44:10: fatal error: crypt.h: No such file or directory
   44 | #include <crypt.h>
      |          ^~~~~~~~~
compilation terminated.
```

### Additional context

Happens on master (commit ee18200c318e7d62f7bdad3a9593b129d251e0db) and unstable (988cc958c57ce4350ec248d2d53087777f9e1949).

### Notify maintainers

@jtojnar @brandonedens @AndersonTorres 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.89-xanmod1, NixOS, 22.11 (Raccoon), 22.11.2523.1b82144edfc`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - channels(bt1cn): `"unstable"`
 - channels(root): `"nixos-22.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by dotlambda**
###### Description of changes
See https://github.com/NixOS/nixpkgs/issues/218108#issuecomment-1455146680.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

closes https://github.com/NixOS/nixpkgs/issues/218108

### CODE REVIEW

**Review by OPNA2608 — APPROVED** [review]
Result of `nixpkgs-review pr 219709` run on x86_64-linux & x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>unrpa (python310Packages.unrpa)</li>
    <li>python311Packages.unrpa</li>
  </ul>
</details>

Can still extract RPA archives on both platforms, LGTM.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 219709` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>unrpa (python310Packages.unrpa)</li>
    <li>python311Packages.unrpa</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
Change looks good to me.

File: pkgs/development/python-modules/unrpa/default.nix [hunk 1]
```diff
@@ -1,22 +1,34 @@
...
+  disabled = pythonOlder "3.7";
 
   src = fetchPypi {
     inherit pname version;
     sha256 = "0yl4qdwp3in170ks98qnldqz3r2iyzil5g1775ccg98qkh95s724";
   };
 
-  propagatedBuildInputs = [ uncompyle6 ];
+  passthru.optional-dependencies = {
+    ZiX = [ uncompyle6 ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/unrpa/default.nix:21]: @dotlambda do we normalise optional dependency names?

Comment (dotlambda) [reply pkgs/development/python-modules/unrpa/default.nix:21]: No idea. I'm not even sure if normalising breaks things there.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/unrpa/default.nix:21]: as far as I know we don't use it anywhere other than nix code, so an eval should catch everything in tree

**Pull request merged**